metadata-ingestion/tests/integration/dbt/test_dbt.py

```diff
@@ -206,8 +221,126 @@ def test_dbt_ingest(pytestconfig, tmp_path, mock_time, **kwargs):
+        "stateful_ingestion": {
+            "enabled": True,
+            "remove_stale_metadata": True,
+            "state_provider": {
+                "type": "datahub",
+                "config": {"datahub_api": {"server": GMS_SERVER}},
+            },
+        },
+    }
+
+    scd_before_deletion: Dict[str, Any] = {
+        "manifest_path": manifest_path,
+        "catalog_path": catalog_path,
+        "sources_path": sources_path,
+        "target_platform": "postgres",
+        "load_schemas": True,
+        # This will bypass check in get_workunits function of dbt.py
+        "write_semantics": "OVERRIDE",
+        # enable stateful ingestion
+        **stateful_config,
+    }
+
+    scd_after_deletion: Dict[str, Any] = {
+        "manifest_path": manifest_path_deleted_actor,
+        "catalog_path": catalog_path_deleted_actor,
+        "sources_path": sources_path_deleted_actor,
+        "target_platform": "postgres",
+        "load_schemas": True,
+        "write_semantics": "OVERRIDE",
+        # enable stateful ingestion
+        **stateful_config,
+    }
+
+    pipeline_config_dict: Dict[str, Any] = {
+        "source": {
+            "type": "dbt",
+            "config": scd_before_deletion,
+        },
+        "sink": {
+            # we are not really interested in the resulting events for this test
+            "type": "console"
+        },
+        "pipeline_name": "statefulpipeline",
+    }
+
+    with patch(
+        "datahub.ingestion.source.state_provider.datahub_ingestion_checkpointing_provider.DataHubGraph",
+        mock_datahub_graph,
+    ) as mock_checkpoint:
+
+        mock_checkpoint.return_value = mock_datahub_graph
+
+        # Do the first run of the pipeline and get the default job's checkpoint.
+        pipeline_run1 = run_and_get_pipeline(pipeline_config_dict)
+        checkpoint1 = get_current_checkpoint_from_pipeline(pipeline_run1)
+
+        assert checkpoint1
+        assert checkpoint1.state
+
+        # Set dbt config where actor table is deleted.
+        pipeline_config_dict["source"]["config"] = scd_after_deletion
+        # Do the second run of the pipeline.
+        pipeline_run2 = run_and_get_pipeline(pipeline_config_dict)
+        checkpoint2 = get_current_checkpoint_from_pipeline(pipeline_run2)
+
+        assert checkpoint2
+        assert checkpoint2.state
+
+        # Perform all assertions on the states. The deleted table should not be
+        # part of the second state
+        state1 = cast(BaseSQLAlchemyCheckpointState, checkpoint1.state)
+        state2 = cast(BaseSQLAlchemyCheckpointState, checkpoint2.state)
+        difference_urns = list(state1.get_table_urns_not_in(state2))
+
+        assert len(difference_urns) == 2
+
+        urn1 = "urn:li:dataset:(urn:li:dataPlatform:dbt,pagila.public.actor,PROD)"
+        urn2 = "urn:li:dataset:(urn:li:dataPlatform:postgres,pagila.public.actor,PROD)"
+
+        assert urn1 in difference_urns
+        assert urn2 in difference_urns
```

Comment:
it'd be good to also assert that the pipeline results from the second run contains the deleted urn with a `Status(deleted=false)` aspect